### ISSUE TITLE
[Bug]: OpenVINO converts PyTorch scaled_dot_product_attention bool mask to NaN output

### ISSUE THREAD

**Issue by ALinrunrun** [openvinotoolkit/openvino#35959]
### OpenVINO Version

2026.1.0

### Operating System

Other (Please specify in description)

### Device used for inference

CPU

### Framework

ONNX

### Model used

Minimal PyTorch model converted by ov.convert_model.

### Issue description

OpenVINO CPU returns `NaN` for a PyTorch model containing `torch.nn.functional.scaled_dot_product_attention` with an all-False boolean attention mask.

For the same input, PyTorch returns a finite all-zero tensor, while the OpenVINO-converted model returns all `NaN`.

OS: Ubuntu 24.04.2 LTS / Linux x86_64

### Step-by-step reproduction

```
import sys
import numpy as np
import openvino as ov
import torch

class M(torch.nn.Module):
    def forward(self, q, k, v, mask):
        return torch.nn.functional.scaled_dot_product_attention(q, k, v, attn_mask=mask)

torch.manual_seed(13)

model = M().eval()

q = torch.randn(1, 1, 3, 4)
k = torch.randn(1, 1, 3, 4)
v = torch.randn(1, 1, 3, 4)
mask = torch.zeros(1, 1, 3, 3, dtype=torch.bool)

torch_out = model(q, k, v, mask).detach().numpy()

ov_model = ov.convert_model(
    model,
    example_input={
        "q": q,
        "k": k,
        "v": v,
        "mask": mask,
    },
)

compiled = ov.Core().compile_model(ov_model, "CPU")

openvino_out = compiled(
    {
        "q": q.numpy(),
        "k": k.numpy(),
        "v": v.numpy(),
        "mask": mask.numpy(),
    }
)[compiled.outputs[0]]

print("q_shape:", tuple(q.shape))
print("mask_all_false:", bool((~mask).all()))
print("torch:", torch_out)
print("openvino_cpu:", openvino_out)
print("torch_finite:", bool(np.isfinite(torch_out).all()))
print("openvino_finite:", bool(np.isfinite(openvino_out).all()))

if np.isfinite(torch_out).all() and not np.isfinite(openvino_out).all():
    print("\nBUG REPRODUCED: OpenVINO returns NaN for converted SDPA bool-mask model.")
    sys.exit(0)

print("\nNOT REPRODUCED")
sys.exit(1)
```

### Relevant log output

```shell
Expected:
OpenVINO CPU should return a finite output consistent with PyTorch for scaled_dot_product_attention with an all-False boolean attention mask.

Actual:
q_shape: (1, 1, 3, 4)
mask_all_false: True
torch: [[[[0. 0. 0. 0.]
   [0. 0. 0. 0.]
   [0. 0. 0. 0.]]]]
openvino_cpu: [[[[nan nan nan nan]
   [nan nan nan nan]
   [nan nan nan nan]]]]
torch_finite: True
openvino_finite: False

BUG REPRODUCED: OpenVINO returns NaN for converted SDPA bool-mask model.
```

### Issue submission checklist

- [x] I'm reporting an issue. It's not a question.
- [x] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [x] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by diego-villalobos** [comment]
Ref. 187128

**Comment by ssam18** [comment]
Reproduced and traced this to softmax of an all minus infinity row when the boolean attn_mask is fully False. PR #36016 adds a fully masked row guard in the decomposition, the reference implementation, and the intel_cpu softmax kernel, plus a regression pytest covering the case from the reproducer.
